cni.go

```diff
@@ -66,84 +70,63 @@ func convertCNIResult(cniResult cniTypes.Result) (NetworkInfo, error) {
 	}
 }
 
-func (n *cni) addVirtInterfaces(pod Pod, networkNS *NetworkNamespace) error {
+func (n *cni) invokePlugins(pod Pod, networkNS *NetworkNamespace) (*NetworkInfo, error) {
 	netPlugin, err := cniPlugin.NewNetworkPlugin()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	for idx, endpoint := range networkNS.Endpoints {
-		virtualEndpoint, ok := endpoint.(*VirtualEndpoint)
-		if !ok {
-			continue
-		}
-
-		result, err := netPlugin.AddNetwork(pod.id, networkNS.NetNsPath, virtualEndpoint.Name())
-		if err != nil {
-			return err
-		}
-
-		netInfo, err := convertCNIResult(result)
-		if err != nil {
-			return err
-		}
-
-		networkNS.Endpoints[idx].SetProperties(netInfo)
+	// Note: In the case of multus or cni-genie this will return only the results
+	// corresponding to the primary interface. The remaining results need to be
+	// derived
+	result, err := netPlugin.AddNetwork(pod.id, networkNS.NetNsPath, CniPrimaryInterface)
+	if err != nil {
+		return nil, err
+	}
 
-		n.Logger().Infof("AddNetwork results %s", result.String())
+	netInfo, err := convertCNIResult(result)
+	if err != nil {
+		return nil, err
 	}
 
-	return nil
+	// We do not care about this for now but
+	// If present, the CNI DNS result has to be updated in resolv.conf
+	// if the kubelet has not supplied it already
+	n.Logger().Infof("AddNetwork results %s", result.String())
+
+	return &netInfo, nil
 }
 
-func (n *cni) deleteVirtInterfaces(pod Pod, networkNS NetworkNamespace) error {
+func (n *cni) deleteInterfaces(pod Pod, networkNS NetworkNamespace) error {
```

Comment:
Now that you have replaced `addVirtInterfaces()` with `invokePlugins()`, I think it would be better to add a parameter to `invokePlugins()` so that you can do both `add` or `remove` action from the same function.
Or I would prefer to have `invokePluginsAdd()` and `invokePluginsDelete()` for more consistency.

Reply:
Yeah, agreed @sboeuf

Reply:
Thx